jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/Generator.kt

```diff
@@ -158,4 +163,81 @@ fun main(args: Array<String>) {
     }
 
     codeRepresentation.dumpTo(projectPath)
+    gradlewAssemble(targetLanguage, projectPath)
+}
+
+private fun gradlewAssemble(targetLanguage: TargetLanguage, projectPath: Path) {
+    val zipName = targetLanguage.projectZipInResourcesName()
+    val workingDir = File(projectPath.toFile(), zipName.substring(0, zipName.length - 4))
+    checkJava(workingDir)
+    if (!isWindows) {
+        chmodGradlew(workingDir)
+    }
+    val gradlewScript = "./gradlew" + if (isWindows) ".bat" else ""
+    runCmd(
+        cmd = listOf(gradlewScript, "assemble"),
+        errorMessage = "problems with assembling",
+        filePrefix = "gradlew",
+        logPrefix = "gradle assembling",
+        workingDir = workingDir
+    )
+}
+
+private fun runCmd(
+    cmd: List<String>,
+    errorMessage: String,
+    filePrefix: String,
+    logPrefix: String,
+    workingDir: File
+): File {
+    val errorFileName = workingDir.absolutePath + File.separator + filePrefix + "Error"
+    val outputFileName = workingDir.absolutePath + File.separator + filePrefix + "Output"
+    val errorFile = File(errorFileName)
+    val outputFile = File(outputFileName)
+    try {
+        val cmdBuilder = ProcessBuilder()
+        cmdBuilder.directory(workingDir)
+        cmdBuilder.command(cmd)
+        val cmdProcess = cmdBuilder.redirectError(errorFile).redirectOutput(outputFile).start()
```

Comment:
Here you chain invocations, one line above do not. Consider consistent invocation contract: either always chain or not. I prefer chaining

Reply:
I believe it is initialize all properties in build initialization site